sbncode/SinglePhotonAnalysis/Libraries/DBSCAN.h

```diff
@@ -0,0 +1,149 @@
+/**
+ * \file DBSCAN.h
+ *
+ * 
+ * \brief Class def header for a class DBSCAN
+ *
+ * @author mark ross-lonergan markrl@nevis.columbia.edu
+ * Written 20th May 2019.
+ */
+
+#ifndef DBSCAN_H
+#define DBSCAN_H
+
+#include <iostream>
+#include <algorithm>
+#include <cmath>
+#include <vector>
+#include <climits>
+#include <limits>
+
+
+class DBSCAN{
+
+  public:
+    double m_eps;
+    int m_minpts;
+
+    /// constructor
+    DBSCAN(double in_eps, int in_minpts): m_eps(in_eps), m_minpts(in_minpts) {}
+
+    /// Default destructor
+    // Guanqun: probably not needed
+    ~DBSCAN(){}
+
+    // scan over points, and group them into different clusters
+    std::vector<int> Scan2D(std::vector<std::vector<double>> &pts);
+    // grab neighbours of point i.
+    std::vector<std::vector<double>> GetNeighbours(size_t i, std::vector<std::vector<double>> &pts,bool);
+    // merge neighbours in seed and pts together into seed
+    int UnionSets(std::vector<std::vector<double>> &seed, std::vector<std::vector<double>> &pts);
+};
+
+std::vector<int> DBSCAN::Scan2D(std::vector<std::vector<double>> &pts){
```

Comment:
Can we move all the implementation from here to the end of the file into a DBSCAN.cxx?